config/v1/types_infrastructure.go

```diff
@@ -1769,9 +1769,9 @@ type NutanixFailureDomain struct {
 	// obtained from the Prism Central console or using the prism_central API.
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MinItems=1
-	// +kubebuilder:validation:MaxItems=1
-	// +listType=map
-	// +listMapKey=type
+	// +kubebuilder:validation:MaxItems=32
+	// +listType=atomic
+	// +kubeubilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="each subnet must be unique"
```

Comment:
Typo in this one sorry
```suggestion
	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="each subnet must be unique"
```

Reply:
Actually, adding the suggested validation rule failed the ci/prow/verify-crd-schema test. When I tried to apply the "infrastructure" crd to my local OCP cluster, I got the same error: 
--------
The CustomResourceDefinition "infrastructures.config.openshift.io" is invalid: 
spec.validation.openAPIV3Schema.properties[spec].properties[platformSpec].properties[nutanix].properties[failureDomains].items.properties[subnets].x-kubernetes-validations[0].rule: Forbidden: estimated rule cost exceeds budget by factor of more than 100x (try simplifying the rule, or adding maxItems, maxProperties, and maxLength where arrays, maps, and strings are declared) ......
-------

So I removed this validation rule.

Reply:
Is it because the parents slice, `FailureDomains` within the `NutanixPlatformSpec` does not have a `maxItems`. Do you know how many possible values we might expect for a list of `FailureDomains`? We could potentially impose a limit retrospectively thanks to ratcheting validation

Reply:
The NutanixMachineProviderConfig can optionally refer to one FailureDomain by name. The FailureDomain configuration is in the Infrastructure CR.

Reply:
With the below extra rule, the ci/prow/verify-crd-schema and all the e2e test runs, and my local test failed with the above mentioned error:  "infrastructures.config.openshift.io" is invalid
Without this rule (the current code change), all the test passed, as well as my local test. 

// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="each subnet must be unique"

Reply:
@JoelSpeed Can you see if the PR is approvable? I hope it can merge today so as to unblock the other 3 dependent PRs review process.

Reply:
> The NutanixMachineProviderConfig can optionally refer to one FailureDomain by name. The FailureDomain configuration is in the Infrastructure CR.

And this field is being added to the infrastructure CR, as I mentioned, you need to limit the field FailureDomains slice that this field exists within, so that you can add the rule back

> @JoelSpeed Can you see if the PR is approvable? I hope it can merge today so as to unblock the other 3 dependent PRs review process.

I previously requested this feature to be put behind a feature gate, I don't know if you saw that?

Reply:
@JoelSpeed Done the changes suggested, and the test ci/prow/verify-crd-schema passed. Please take another look.

Reply:
Added the feature gate as required.

Reply:
@JoelSpeed After adding the feature gate NutanixMultiSubnets, the ci/prow/verify-crd-schema  test run failed with "error in machineconfiguration/v1/zz_generated.featuregated-crd-manifests/controllerconfigs.machineconfiguration.openshift.io/NutanixMultiSubnets.yaml". The e2e-test runs and my manual test passed. Can you take a look? I don't know what caused these errors.